integration-tests-ex/cases/src/test/java/org/apache/druid/testsEx/msq/ITMSQForBatchIndex.java

```diff
@@ -0,0 +1,144 @@
+public class ITMSQForBatchIndex
+{
+  public static final Logger LOG = new Logger(TestQueryHelper.class);
+  @Inject
+  private MsqTestQueryHelper msqHelper;
+
+  @Inject
+  protected TestQueryHelper queryHelper;
+
+  @Inject
+  private IntegrationTestingConfig config;
+
+  @Inject
+  private ObjectMapper jsonMapper;
+
+  @Inject
+  private DataLoaderHelper dataLoaderHelper;
+
+  @Inject
+  private CoordinatorResourceTestClient coordinatorClient;
+
+  private static final String BATCH_INDEX_TASKS_DIR = "/multi-stage-query/batch-index/";
+
+  protected void doTestQuery(String dataSource, String queryFilePath)
+  {
+    try {
+      String queryResponseTemplate;
+      try {
+        InputStream is = AbstractITBatchIndexTest.class.getResourceAsStream(queryFilePath);
+        queryResponseTemplate = IOUtils.toString(is, StandardCharsets.UTF_8);
+      }
+      catch (IOException e) {
+        throw new ISE(e, "could not read query file: %s", queryFilePath);
+      }
+
+      queryResponseTemplate = StringUtils.replace(
+        queryResponseTemplate,
+        "%%DATASOURCE%%",
+        dataSource
+      );
+      queryHelper.testQueriesFromString(queryResponseTemplate);
+
+    }
+    catch (Exception e) {
+      LOG.error(e, "Error while testing");
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testMsqIngestionForBatchIndexTasks() throws Exception
+  {
+    File[] files = (new File(getClass().getResource(BATCH_INDEX_TASKS_DIR).toURI())).listFiles();
+    LOG.info(Arrays.toString(files));
+    String datasource = "dst";
+    for (int i = 1; i < files.length; i++) {
+      LOG.info("Starting MSQ test for [%s]", files[i]);
+      MsqSqlTasksWithTestQueries batchSqlTasksAndQueries =
+          jsonMapper.readValue(
+              TestQueryHelper.class.getResourceAsStream(BATCH_INDEX_TASKS_DIR + files[i].getName()),
+              new TypeReference<MsqSqlTasksWithTestQueries>()
+              {
+              }
+          );
+      // Clear up the datasource from the previous runs
+      coordinatorClient.unloadSegmentsForDataSource(datasource + files[i].getName());
+
+      String sqlTask = StringUtils.replace(
+          batchSqlTasksAndQueries.getSqlTask(),
+          "%%DATASOURCE%%",
+          datasource + files[i].getName()
+      );
+
+      LOG.info("SqlTask - \n %s", sqlTask);
+
+      // Submit the tasks and wait for the datasource to get loaded
+      SqlTaskStatus sqlTaskStatus = msqHelper.submitMsqTask(sqlTask);
+
+      LOG.info("Sql Task submitted with task Id - %s", sqlTaskStatus.getTaskId());
+
+      if (sqlTaskStatus.getState().isFailure()) {
+        Assert.fail(StringUtils.format(
+              "Unable to start the task successfully.\nPossible exception: %s",
+              sqlTaskStatus.getError()
+        ));
+      }
+      msqHelper.pollTaskIdForCompletion(sqlTaskStatus.getTaskId());
+      dataLoaderHelper.waitUntilDatasourceIsReady(datasource + files[i].getName());
```

Comment:
I think this should be pushed inside msqHelper or a base test class. It would most likely get copied in every test case.